src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -14,13 +18,62 @@
   "v3")
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Private
+
+(defn class-info-from-jruby->class-info-for-json
+  "Convert a class info map received from the jruby service into an
+  appropriate map for use in serializing an environment_classes response to
+  JSON"
+  [info-from-jruby environment]
+  (->> info-from-jruby
+       (map #(hash-map :path (key %) :classes (val %)))
+       (sort-by :path)
+       (into [])
+       (hash-map :name environment :files)))
```

Comment:
So this fn is the entirety of the munging you need to do in order to go from the Ruby API to the wire format?